src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -902,14 +902,9 @@ private TypeSymbol BindVariableType(CSharpSyntaxNode declarationNode, Diagnostic
 
             if (kind == LocalDeclarationKind.FixedVariable || kind == LocalDeclarationKind.UsingVariable)
             {
-                // CONSIDER: The error message is "you must provide an initializer in a fixed 
-                // CONSIDER: or using declaration". The error message could be targetted to 
-                // CONSIDER: the actual situation. "you must provide an initializer in a 
-                // CONSIDER: 'fixed' declaration."
-
                 if (initializerOpt == null)
                 {
-                    Error(localDiagnostics, ErrorCode.ERR_FixedMustInit, declarator);
+                    Error(localDiagnostics, ErrorCode.ERR_FixedMustInit, declarator, kind == LocalDeclarationKind.FixedVariable ? "fixed" : "using");
```

Comment:
SyntaxFacts.GetText doesn't accept a LocalDeclarationKind. Is there anything else I could use for this?